web/src/narrow_state.ts

```diff
@@ -131,6 +131,8 @@ export let stream_id = (current_filter: Filter | undefined = filter()): number |
     }
     const stream_operands = current_filter.operands("channel");
     if (stream_operands.length === 1 && stream_operands[0] !== undefined) {
+        // If the operand is not a valid channel ID, then
+        // it is an empty string, and this will return NaN.
         return Number.parseInt(stream_operands[0], 10);
```

Comment:
I think it'd be better to only have one class of invalid value coming out of this function -- let's just have this do the `NaN` check and replace it with returning `undefined`.

Reply:
Done. Added that as a separate commit in these changes.